source/plugin/modularization.rst

```diff
@@ -0,0 +1,62 @@
+===================
+Code Modularization
+===================
+
+When writing Sponge plugins, you may find that you need to modularize your code. Maybe you want to split your core into
+another project so that others can make use of it. Maybe you want to utilize a service another plugin provides. Or maybe
```

Comment:
This would look better with fewer sentences. eg. "Maybe you want to [blah blah], or utilize a service [blah]". The fourth sentence is totally unnecessary, and starting with a preposition often looks bad anyway.